### DISCUSSION TITLE
how to use querys of arbitrary dicts,   like Bodies of arbitrary dicts?

### DISCUSSION THREAD

**Discussion by xingshuyin** [fastapi/fastapi#6207] [Questions] [answered]
### First Check

- [ ] I added a very descriptive title to this issue.
- [ ] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [ ] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.get("/get_list")
def get_list(q:Dict[str,Union[str,int,float]]):
    return q
```

### Description

https://fastapi.tiangolo.com/tutorial/body-nested-models/#bodies-of-arbitrary-dicts

l got this from official website, but is only for request body,  i want something like this for query

### Operating System

Windows

### FastAPI Version

0.88.0

### Python Version

3.11

**Comment by TheJumpyWizard** [comment]
Could you clarify what exactly you are asking? I see the code example you provided, but what do you mean how to use queries of arbitrary dicts? I have found a few different resources I believe would be useful for you but want to clarify what you are asking before I link them. @xingshuyin

**Comment by xingshuyin** [comment]
> 

don`t care about example ,the Description is all the question

**Comment by xingshuyin** [comment]
> Could you clarify what exactly you are asking? I see the code example you provided, but what do you mean how to use queries of arbitrary dicts? I have found a few different resources I believe would be useful for you but want to clarify what you are asking before I link them. @xingshuyin

i wang to just use one position argument receive all query params, because i have many query ,and i dont want to declare all of them

**Comment by davismartens** [comment]
> > Could you clarify what exactly you are asking? I see the code example you provided, but what do you mean how to use queries of arbitrary dicts? I have found a few different resources I believe would be useful for you but want to clarify what you are asking before I link them. @xingshuyin
> 
> i wang to just use one position argument receive all query params, because i have many query ,and i dont want to declare all of them

If you want to pass them all in without declaring them, why don't you just accept them as dict in the body?

```
@app.post("/foo/")
def foo(data: dict):
    return data

```
Unless you declare them or use a schema, you can't really enforce their type so no need to specify them in your endpoint.

**Comment by xingshuyin** [comment]
> 

when I use restful api to get list of item.  I have to use get method, so I have to declare filter params in query instead of body

**Comment by jgould22** [comment, marked as answer]
If you dont want any validation or anything you can just get the query params from the request object
```python
from fastapi import  Request
@app.get("/get_list")
def get_list(request: Request):
     q = request.query_params
    return q
```

**Comment by xingshuyin** [comment]
> If you dont want any validation or anything you can just get the query params from the request object
> 
> ```python
> from fastapi import  Request
> @app.get("/get_list")
> def get_list(request: Request):
>      q = request.query_params
>     return q
> ```

ok ，that's I wanted thank you very much
